packages/beacon-node/src/chain/validation/attestation.ts

```diff
@@ -514,27 +513,26 @@ async function validateAttestationNoSignatureCheck(
   }
 
   // no signature check, leave that for step1
-  const indexedAttestationContent = {
+  const indexedAttestation: IndexedAttestation = {
     attestingIndices,
     data: attData,
     signature,
   };
-  const indexedAttestation =
-    ForkSeq[fork] >= ForkSeq.electra
-      ? (indexedAttestationContent as electra.IndexedAttestation)
-      : (indexedAttestationContent as phase0.IndexedAttestation);
 
-  const attestationContent = attestationOrCache.attestation ?? {
-    aggregationBits,
-    data: attData,
-    committeeIndex,
-    signature,
-  };
-
-  const attestation =
-    ForkSeq[fork] >= ForkSeq.electra
-      ? (attestationContent as SingleAttestation<ForkPostElectra>)
-      : (attestationContent as SingleAttestation<ForkPreElectra>);
+  const attestation: SingleAttestation = attestationOrCache.attestation
+    ? attestationOrCache.attestation
```

Comment:
You can use `??` here to avoid `attestationOrCache.attestation` twice

Reply:
yes, but unclear if that makes it more readable since you need parentheses around the ternary operation
```ts
  const attestation: SingleAttestation =
    attestationOrCache.attestation ??
    (!isForkPostElectra(fork)
      ? {
          aggregationBits: aggregationBits as BitArray,
          data: attData,
          signature,
        }
      : {
          committeeIndex,
          attesterIndex: validatorIndex,
          data: attData,
          signature,
        });
```
if-else is not nice either imo